benchmarktests/target_secret_azure.go

```diff
@@ -0,0 +1,180 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package benchmarktests
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/go-uuid"
+	"github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/hcl/v2/gohcl"
+	"github.com/hashicorp/vault/api"
+	vegeta "github.com/tsenart/vegeta/v12/lib"
+)
+
+// Constants for test
+const (
+	AzureSecretTestType       = "azure_secret"
+	AzureSecretTestMethod     = "GET"
+	AzureSecretSubscriptionID = VaultBenchmarkEnvVarPrefix + "SUBSCRIPTION_ID"
+	AzureSecretTenantID       = VaultBenchmarkEnvVarPrefix + "TENANT_ID"
+	AzureSecretClientID       = VaultBenchmarkEnvVarPrefix + "CLIENT_ID"
+	AzureSecretClientSecret   = VaultBenchmarkEnvVarPrefix + "CLIENT_SECRET"
+	AzureSecretEnvironment    = VaultBenchmarkEnvVarPrefix + "ENVIRONMENT"
+)
+
+func init() {
+	// "Register" this test to the main test registry
+	TestList[AzureSecretTestType] = func() BenchmarkBuilder { return &AzureTest{} }
+}
+
+type AzureTest struct {
+	pathPrefix string
+	header     http.Header
+	roleName   string
+	config     *AzureSecretTestConfig
+	logger     hclog.Logger
+}
+
+type AzureSecretTestConfig struct {
+	AzureConfig *AzureSecretConfig `hcl:"azure,block"`
+	AzureRole   *AzureSecretRole   `hcl:"role,block"`
+}
+
+type AzureSecretConfig struct {
+	SubscriptionId  string `hcl:"subscription_id,optional"`
+	TenantId        string `hcl:"tenant_id,optional"`
+	ClientId        string `hcl:"client_id,optional"`
+	ClientSecret    string `hcl:"client_secret,optional"`
+	Environment     string `hcl:"environment,optional"`
+	PasswordPolicy  string `hcl:"password_policy,optional"`
+	RootPasswordTTL string `hcl:"root_password_ttl,optional"`
+}
+
+type AzureSecretRole struct {
+	Name                string `hcl:"name,optional"`
+	AzureRoles          string `hcl:"azure_roles,optional"`
+	AzureGroups         string `hcl:"azure_groups,optional"`
+	ApplicationObjectId string `hcl:"application_object_id,optional"`
+	PersistApp          bool   `hcl:"persist_app,optional"`
+	TTL                 string `hcl:"ttl,optional"`
+	MaxTTL              string `hcl:"max_ttl,optional"`
+	PermanentlyDelete   bool   `hcl:"permanently_delete,optional"`
+}
+
+func (a *AzureTest) ParseConfig(body hcl.Body) error {
+	testConfig := &struct {
+		Config *AzureSecretTestConfig `hcl:"config,block"`
+	}{
+		Config: &AzureSecretTestConfig{
+			AzureConfig: &AzureSecretConfig{
+				SubscriptionId: os.Getenv(AzureSecretSubscriptionID),
+				TenantId:       os.Getenv(AzureSecretTenantID),
```

Comment:
These two fields are required for configuration. We should have a check below to make sure that we actually have a value and not `""` and allow to fail out early and provide a meaningful error, prior to creating any resources in Vault.